consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/IbftGossip.java

```diff
@@ -93,16 +95,20 @@ public boolean gossipMessage(final Message message) {
         throw new IllegalArgumentException(
             "Received message does not conform to any recognised IBFT message structure.");
     }
-    final Signature signature = signedData.getSignature();
-    if (seenMessages.contains(signature)) {
+    final List<Address> excludeAddressesList =
+        Lists.newArrayList(message.getConnection().getPeer().getAddress(), signedData.getSender());
+
+    return send(messageData, excludeAddressesList);
+  }
+
+  @Override
+  public boolean send(final MessageData messageData, final List<Address> excludeAddressesList) {
+    Hash uniqueID = Hash.hash(messageData.getData());
```

Comment:
Could we just use the hashCode from the messageData instead to uniquely identify the message?